armi/reactor/reactors.py

```diff
@@ -2316,3 +2326,12 @@ def processLoading(self, cs, dbLoad: bool = False):
         self.p.maxAssemNum = self.getMaxParam("assemNum")
 
         getPluginManagerOrFail().hook.onProcessCoreLoading(core=self, cs=cs)
+
+    def updateBlockBOLHeights(self):
+        """post thermal expansion, update block BOL heights"""
+        self.p.referenceBlockAxialMesh = self.findAllAxialMeshPoints(applySubMesh=False)
+        self.p.axialMesh = self.findAllAxialMeshPoints()
```

Comment:
why does the mesh setting code need to run again?

Reply:
It is handled pre-expansion, yes. But I think it should be updated. `findAllAxialMeshPoints` ends up in `findAllMeshPoints` and this recalculates mesh positions based on `b.spatialLocator.getGlobalCellBase()` and `b.spatialLocator.getGlobalCellTop()` which change after expansion.

Reply:
great point